Restore homogeneous dataset collection indicator

davelopez opened the pull request:
Fixes #13744

<del>I had to move the `elements_datatype` serialization field from the `detailed` to the `summary` view so it can be rendered in the history items without making additional requests. Hope that's ok.</del> See https://github.com/galaxyproject/galaxy/pull/13772#issuecomment-1106246781

Also, I slightly changed the description string (compared to the old history) depending on the `collection_type`, for example:
`list:paired` (heterogeneous): *a list of pairs with 10 items*  -> *a list with 10 pairs*
`list:paired` (homogeneous): *a list of pairs with 10 items*  -> *a list with 10 **tabular** pairs*

There is a complete [list of samples in the tests](https://github.com/galaxyproject/galaxy/compare/dev...davelopez:restore_homogeneous_hdca_indicator?expand=1#diff-a68e5baf968484751bfe46e55652d7ae3fa46c57a4abd21b08d5a8f3abcec32eR19-R58). I'm happy to further tweak this if neccesary.

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] Instructions for manual testing are as follows:
  - Create some collections with the same (and different) types of datasets
  - Observe the collection description matches the contents

![Screenshot from 2022-04-20 14-55-55](https://user-images.githubusercontent.com/46503462/164240005-66d6c678-5a4e-4334-9033-c3e62cc1a44a.png)

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

davelopez committed:
Move `elements_datatypes` to summary view

Since the new history doesn't get the detailed collection when retrieving contents I had to move this serializable property to the summary view

davelopez committed:
Display homogeneous collection types on description

davelopez committed:
Remove unused JSON test data

mvdbeek commented on client/src/components/History/Content/Collection/CollectionDescription.test.js:
```diff
@@ -11,23 +11,50 @@ describe("CollectionDescription", () => {
+
         await wrapper.setProps({ collectionType: "other" });
-        expect(details.at(0).text()).toBe("a nested list");
+        expect(wrapper.text()).toBe(`a nested list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} datasets`);
```
These aren't datasets, `other` would be a deeply nested collection.
```suggestion
        expect(wrapper.text()).toBe(`a nested list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} dataset collections`);
```

davelopez replied:
Thanks for the catch Marius, I've rebased with the changes.

davelopez committed:
Consider any other collection_type as nested collection

In the collection description

davelopez committed:
Revert "Move `elements_datatypes` to summary view"

This reverts commit eccb0c8e9d59363d068e47f6b5938bd89bf20bb5.

davelopez committed:
Conditionally include `elements_datatypes` when serializing collections

Only requests with a particular media type (used by the history UI) will include this additional field for dataset collections.

mvdbeek reviewed (approved):
Very nice!
